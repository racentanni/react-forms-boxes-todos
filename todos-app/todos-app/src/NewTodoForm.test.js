import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders without crashing', () => {
  render(<NewTodoForm addTodo={() => {}} />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test('calls addTodo on form submission', () => {
  const addTodo = jest.fn();
  const { getByPlaceholderText, getByText } = render(<NewTodoForm addTodo={addTodo} />);
  const input = getByPlaceholderText('New Todo');
  const button = getByText('Add Todo');

  fireEvent.change(input, { target: { value: 'Test Todo' } });
  fireEvent.click(button);

  expect(addTodo).toHaveBeenCalledWith('Test Todo');
});