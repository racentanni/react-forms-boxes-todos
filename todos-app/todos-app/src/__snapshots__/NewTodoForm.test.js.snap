// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="New Todo"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
